ts/main.ts

```diff
@@ -147,6 +161,92 @@ if (!$monsterPage) throw new Error('Could not load Monster Page');
 $monsterPage.addEventListener('click', function () {
   $monLine.classList.remove('hidden');
   $monSelect.classList.add('hidden');
+  $monFav.classList.add('hidden');
+  $favoriteTab?.classList.remove('selected');
+  $monsterPage.classList.add('selected');
+  $favTitle?.classList.add('hidden');
+  $monTitle?.classList.remove('hidden');
+  $filter.classList.remove('hidden');
+  $star.classList.add('hidden');
+  $star?.classList.remove('favorite');
+});
+
+if (!$estrella) throw new Error('could not load favorite star click');
+$estrella.addEventListener('click', function () {
+  if ($star.classList[1] === 'favorite') {
+    removeFav();
+  } else if ($star.classList[1] !== 'favorite') {
+    favMon();
+  }
+});
+
+function removeFav(): void {
+  const $card = document.querySelector('.monster-card');
+  if (!$nameMon) throw new Error('Could not load name');
+  if (!$card) throw new Error('Could not load name');
+  if (!$monFavCon?.firstChild)
+    throw new Error('could not load the first child element');
+  if (!$nameMon.textContent) throw new Error('Could not load name');
+  $star?.classList.remove('favorite');
+  for (let i = 0; i < data.favorite.length; i++) {
+    if (Number($nameMon.id) === data.favorite[i].id) {
+      data.favorite.splice(i, 1);
+    }
+  }
+  writeData();
+  if (!$monFav) throw new Error('could not load monster fav page');
+  const $cardToRemove = $monFav.querySelector(
+    `[data-monster-id='${Number($nameMon.id)}']`,
+  );
+  $cardToRemove?.remove();
+}
+
+function favMon(): void {
+  if (!$nameMon) throw new Error('Could not load name');
+  if (!$nameMon.textContent) throw new Error('Could not load name');
+  for (let i = 0; i < data.favorite.length; i++) {
+    if ($nameMon.textContent === data.favorite[i].name) {
+      return;
+    }
+  }
+  $star?.classList.add('favorite');
+  data.favorite.push({
+    name: $nameMon.textContent,
+    photoURL: `images/icons/${$nameMon.textContent}.webp`,
+    id: Number($nameMon.id),
+  });
+  writeData();
+  readData();
+
+  loader();
+}
```

Comment:
You should be able to remote this `readData` call. Also rather than the loader function, you should just render and append the monster card since it's just the one. You would want to save the object to a variable first though.
```
const favMon = {
  name: $nameMon.textContent,
  photoURL: `images/icons/${$nameMon.textContent}.webp`,
  id: Number($nameMon.id),
}

data.favorite.push(favMon);
$monFavCon.append(renderMonster(favMon));
```